### ISSUE TITLE
HTTP Proxy integration to Lambda is incorrect.

### ISSUE THREAD

**Issue by gantzm** [awslabs/aws-sam-cli#244]
Using a template like the following (see below) the "resource" attribute of the proxy JSON should be set to `/rest/getId/{token}`.  

When issuing a GET call to `/rest/getId/MyId`, I would expect to see:

```
{
  "resource": "/rest/getId/{token}",
  "path": "/rest/getId/MyId"
}
```
Instead I receive the following:
```
{
  "resource": "/rest/getId/MyId",
  "path": "/rest/getId/MyId"
}
```
This is contrary to how HTTP Proxy works with API Gateway. 

Sample Template:

```
Resources:
  MyResource:
    Type: AWS::Serverless::Function
    Properties:
      Handler: com.MyHandler
      CodeUri: MyHandler.zip
      Runtime: java8
      MemorySize: 512
      Timeout: 30
      Role:
        Fn::ImportValue:
          !Join ['-', [!Ref 'ProjectId', !Ref 'AWS::Region', 'LambdaTrustRole']]
      Events:
        GetId:
          Type: Api
          Properties:
            Path: /rest/getId/{token}
            Method: get
```

**Comment by gantzm** [comment]
I've spent some time trying to fix this myself but have encountered some difficulties.  First, this is my first adventure into Go so I'm having a few issues groking the language itself.  Second, it looks like to fix this I need to somehow get the "resource" property copied into the handler as the handler doesn't retain a reference back to the configuration data.

It's this layer of abstraction that I'm having some issues with.

**Comment by fdesoye** [comment]
+1

**Comment by kind3r** [comment]
Also missing `requestContext.apiId`
Should I add a separate issue ?

**Comment by valtlfelipe** [comment]
I'm missing `requestContext.apiId` too, without it, vandium does not work

**Comment by jfuss** [comment]
The initial issue @gantzm reported has been corrected in the [0.3.0 release](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.3.0). Closing since this has been addressed.

@kind3r and @valtlfelipe `requestContext.apiId` was added to the event and given a default value in the release 0.3.0 as well. Code can be found here: https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/local/events/api_event.py#L67
